docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+
+- spec.destNamespace: A string holding the name of the namespace on destination cluster where imagestreams should be migrated.
+
+- spec.imageStreamRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the imagesstream is present.
+
+
+- spec.srcMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the operator is installed.
+
+     
+## Direct Volume Migration
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: DirectVolumeMigration
+metadata:
+  name: direct1
+  namespace: openshift-migration
+spec:
+  createDestinationNamespaces: false
+  deleteProgressReportingCRs: false
+  destMigClusterRef:
+    name: host
+    namespace: openshift-migration
+  persistentVolumeClaims:
+  - name: pvc-0
+    namespace: pvc-migrate-bmark-1
+  - name: pvc-0
+    namespace: pvc-migrate-bmark-2
+  - name: pvc-0
+    namespace: pvc-migrate-bmark-3
+  - name: pvc-1
+    namespace: pvc-migrate-bmark-3
+  srcMigClusterRef:
+    name: ocp3
+    namespace: openshift-migration
+```
+
+- spec.createDestinationNamespaces: A boolean flag that is set True to create namespaces in destination cluster.
+
+- spec.deleteProgressReportingCRs: A boolean flag to specify if progress reporting CRs needs to be deleted or not.
+
+- spec.destMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the destination mig cluster where the operator is installed.
+
+
+- spec.persistentVolumeClaims: A list of all the PVCs that are to be migrated with direct volume migration.
+
+- spec.srcMigClusterRef: ObjectReference contains enough information to let you inspect or modify the referred object. Contains name and namespace of the source mig cluster where the operator is installed.
+
+
+## Direct Volume Migration Progress
+
+```
+apiVersion: migration.openshift.io/v1alpha1
+kind: DirectVolumeMigrationProgress
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: directvolumemigrationprogress-sample
+spec:
+  # [!] Change clusterRef to point to the cluster where the pod is running
+  clusterRef:
+    name: sample-source-3110
+    namespace: openshift-migration
+  # [!] Change podRef to point to the name and namespace of the running pod
+  podRef:
+    name: directmigration-sample-pod-0
+    namespace: openshift-migration
+```
+
+- spec.clusterRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.podRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+
+## Mig Analytics
```

Comment:
We have not documented MigAnalytics in MTC. When will it be used?

Reply:
@apinnick MigAnalytics gets created when we create a MigPlan explicitly via UI, it is used to gather information regarding K8s resources, images, persistent volumes etc.